applications/RomApplication/custom_strategies/lspg_rom_builder_and_solver.h

```diff
@@ -0,0 +1,478 @@
+     * which will be asked and set by another strategy object
+     */
+
+    ///@name Type Definitions
+    ///@{
+
+    // Class pointer definition
+    KRATOS_CLASS_POINTER_DEFINITION(LSPGROMBuilderAndSolver);
+
+    // The size_t types
+    typedef std::size_t SizeType;
+    typedef std::size_t IndexType;
+
+    /// The definition of the current class
+    typedef LSPGROMBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> ClassType;
+
+    /// Definition of the classes from the base class
+    typedef BuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+    typedef typename BaseType::TSchemeType TSchemeType;
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+    typedef typename BaseType::ElementsArrayType ElementsArrayType;
+    typedef typename BaseType::ConditionsArrayType ConditionsArrayType;
+
+    /// Additional definitions
+    typedef typename ModelPart::MasterSlaveConstraintContainerType MasterSlaveConstraintContainerType;
+    typedef Element::EquationIdVectorType EquationIdVectorType;
+    typedef Element::DofsVectorType DofsVectorType;
+    typedef boost::numeric::ublas::compressed_matrix<double> CompressedMatrixType;
+
+    // Non-distributed, dense:
+    typedef LocalSystemMatrixType RomSystemMatrixType;
+    typedef LocalSystemVectorType RomSystemVectorType;
+
+    //Distributed, dense
+    typedef RomSystemMatrixType LSPGSystemMatrixType; 
+    typedef RomSystemVectorType LSPGSystemVectorType;
+    //      ^ Change this to a distributed dense type
+
+    /// DoF types definition
+    typedef Node<3> NodeType;
+    typedef typename NodeType::DofType DofType;
+    typedef typename DofType::Pointer DofPointerType;
+    typedef moodycamel::ConcurrentQueue<DofType::Pointer> DofQueue;
+
+    ///@}
+    ///@name Life cycle
+    ///@{
+
+    explicit LSPGROMBuilderAndSolver(
+        typename TLinearSolver::Pointer pNewLinearSystemSolver,
+        Parameters ThisParameters)
+        : ROMBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>(pNewLinearSystemSolver, ThisParameters) 
+    {
+        // Validate and assign defaults
+        // Parameters this_parameters_copy = ThisParameters.Clone();
+        // this_parameters_copy = this->ValidateAndAssignParameters(this_parameters_copy, this->GetDefaultParameters());
+        // this->AssignSettings(this_parameters_copy);
+    }
+
+    ~LSPGROMBuilderAndSolver() = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetUpDofSet(

```

Comment:
Is this method diferent from ROMBuilderAndSolver::SetUpDofSet? If it is not, I'd simply remove this version.